### PR TITLE
Anonymize patron data

### PR DESCRIPTION
**Pull request by jimchamp**
Creates the means for an admin to anonymize a patron's account.  This PR can be moved from draft when the following tasks are complete:
- [X] Add "Anonymize Account" button to `/admin/people` view
- [X] Create testable methods for updating and deleting by username
- [x] Delete or otherwise reset the patron's profile page
- [x] Remove patron from all usergroups
- [x] Reset patron's preferences to default 
- [x] Create unit tests for new DB model methods
- [x] Disable `test` flag for anonymize function (no updates/deletions are committed while `test` is true)

### Technical

The `/admin/people/view.html` template was producing invalid HTML.  The initial commit of this PR solely addresses those issues.

__Important Note:__ If the anonymization process throws an error before it completes we can re-run, but some operations may not be possible during the re-run.  This may result in some lingering records.

### Screenshot

admin_people_new_view
_New button in `/admin/people`._

anon_success_flash_message
_Flash message that is displayed on success_

### Stakeholders

@mekarpeles

### CODE REVIEW

**Comment by jimchamp** [comment]
Keeping this in draft, as it doesn't yet address some things:
1. What happens if a new patron is assigned an anonymized patrons old username?  Is this even possible?
2. If #1 _is_ possible, what do we do about the history table on the profile page?
3. Admins can still log in as the anonymized patron and make edits etc. that will be linked to the old username. This shouldn't be an issue, but I wanted to call it out anyway.

**Comment by jimchamp** [comment]
Add dialog or something to confirm action.

**Comment by jimchamp** [comment]
`anonymize` function now removes account information from the store.  Admin UI has the following updates:
1. The anonymize button has been moved to the bottom of the account history table.
2. Anonymization POST is blocked by a confirmation dialog.

Screenshot from 2022-05-27 13-48-02

Running into issues anonymizing keys, which is how we planned on anonymizing lists and edit history.  If it turns out that changes keys is not possible, maybe it would be best to change ownership of lists and history to a dummy account the represents anonymized patrons? Either way, suggest adding this functionality in a PR to be opened in the near future.

File: openlibrary/templates/admin/people/view.html [hunk 1]
```diff
@@ -214,9 +213,24 @@ <h1><a class="plain" href="/admin/people">People</a> / $person.username</h1>
...
     </tbody>
   </table>
 
+<script>
+    function confirmAnonymize(event) {
+        if(!confirm('Really anonymize this patron? This will delete the patron\'s profile page and booknotes, and anonymize the patron\'s reading log, reviews, and star ratings.')) {
+            event.preventDefault()
+        }
+    }
+</script>
```
Comment (jimchamp) [inline openlibrary/templates/admin/people/view.html:233]: If we're happy with the confirmation dialog approach, I can move this inline JS to a `.js` file.

**Comment by jimchamp** [comment]
Today's changes:
1. Moved inline JS to dedicated JS file.
2. Display patron's name in the confirm dialog.
3. Add "Dry run" checkbox, which allows `anonymize` to be executed in test mode.
4. Update reading log visibility to private, which will redirect folks to the deleted profile page when any reading log page is visited.
5. Avoid fetching reading log counts for anonymized patrons.

File: openlibrary/templates/admin/people/view.html [hunk 2]
```diff
@@ -216,22 +216,16 @@ <h1><a class="plain" href="/admin/people">People</a> / $person.username</h1>
     <tr>
         <td>$_("Anonymize Account:")</td>
         <td>
-            <form method="POST" id="anonymize-form">
-                <button type="submit" name="action" value="anonymize_account" style="float: right;" onclick="confirmAnonymize(event)">$_("Anonymize Account")</button>
+            <form method="POST" id="anonymize-form" action="?debug=true">
+                <input type="checkbox" id="dry-run-checkbox" name="dry_run">
+                <label for="dry-run-checkbox">Dry Run</label>
+                <button type="submit" name="action" value="anonymize_account" style="float: right;" class="account-anonymization-button" data-display-name="$person.username">$_("Anonymize Account")</button>
```
Comment (mekarpeles) [inline openlibrary/templates/admin/people/view.html:222]: inline styles?

**Pull request merged**